### ISSUE TITLE
multiline-comment-style: add an option to only enforce asterisks on multiline comments (while allowing single line comments)

### ISSUE THREAD

**Issue by jakub-g** [eslint/eslint#11602]
**What rule do you want to change?**

https://eslint.org/docs/rules/multiline-comment-style

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

new option `"starred-block-or-separate-lines"`

**Please provide some example code that this change will affect:**

```js
/* This
is multiline comment
without star at each line
which is not so great */
const solution = 42

// Line comments are really okay
// Even if they sometimes take multiple lines
const alternativeSolution = 888
```

**What does the rule currently do for this code?**

"starred-block" catches the multiline comment issue, but also prevents single line comments.

**What will the rule do after it's changed?**

It will enforce the multiline comment to be formatted, while allowing the line comments to be kept intact:

```js
/*
 * This
 * is a multiline comment
 * without stars at each line
 * which looks awful
 */
const solution = 42

// Line comments are really okay
// Even if they sometimes take multiple lines
const alternativeSolution = 888
```

**Are you willing to submit a pull request to implement this change?**

yes
